polyfills/Element/prototype/dataset/detect.js

```diff
@@ -0,0 +1,6 @@
+(document.documentElement.dataset &&
+	(
+		Object.getOwnPropertyDescriptor(Element.prototype, 'dataset')  &&
+		Object.getOwnPropertyDescriptor(Element.prototype, 'dataset').get
```

Comment:
hmm, I don't know that that's universally true, eg. for RUM beacon code... https://polyfill.io/v2/polyfill.js?rum=1&ua=chrome/50 (see top section).  We will probably want to close that loophole (and figure out if there are any others), or ensure that detects are always ES3 compatible.

One thing that could be done in the RUM case is putting each detect in a try...catch, but that will bloat the code a fair bit.